sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -321,4 +323,125 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
       });
     }
   }
+
+  /**
+   * Get first outstanding student assessment to be calculated
+   * in the order of original assessment study start date.
+   * @param studentId student id.
+   * @param programYearId program year id.
+   * @returns student assessment to be calculated.
+   */
+  async getOutstandingAssessmentsForStudentInSequence(
+    studentId: number,
+    programYearId: number,
+  ): Promise<StudentAssessmentDetail> {
+    const originalAssessmentStudyStartDateAlias =
+      "originalAssessmentStudyStartDate";
+    // Sub query to get the original assessment study start date of a given assessment's application.
+    const originalAssessmentDateSubQuery = this.repo
+      .createQueryBuilder("studentAssessment")
+      .select("offering.studyStartDate")
+      .innerJoin("studentAssessment.offering", "offering")
+      .where("studentAssessment.application.id = assessment.application.id")
+      .andWhere("studentAssessment.triggerType = :triggerType")
+      .limit(1)
+      .getSql();
+
+    return (
+      this.repo
+        .createQueryBuilder("assessment")
+        .select("assessment.id", "id")
+        .addSelect(
+          `(${originalAssessmentDateSubQuery})`,
+          originalAssessmentStudyStartDateAlias,
+        )
+        .innerJoin("assessment.application", "application")
+        .where(
+          "assessment.studentAssessmentStatus NOT IN (:...studentAssessmentStatus)",
+        )
+        // Exclude the assessments which are waiting for PIR confirmation
+        // as they do not have a confirmed study start date.
+        .andWhere("assessment.offering IS NOT NULL")
+        .andWhere("application.student.id = :studentId")
+        .andWhere("application.programYear.id = :programYearId")
+        .setParameter("triggerType", AssessmentTriggerType.OriginalAssessment)
+        .setParameter("studentAssessmentStatus", [
+          StudentAssessmentStatus.Completed,
+          StudentAssessmentStatus.CancellationRequested,
+          StudentAssessmentStatus.CancellationQueued,
+          StudentAssessmentStatus.Cancelled,
+        ])
+        .setParameter("studentId", studentId)
+        .setParameter("programYearId", programYearId)
+        .orderBy(`"${originalAssessmentStudyStartDateAlias}"`)
+        .addOrderBy("assessment.createdAt")
+        .limit(1)
+        .getRawOne<StudentAssessmentDetail>()
```

Comment:
Do we need to `limit(1)` if we user `getRawOne()`?

Reply:
Yes, mainly with query builder I'm not sure about the order of first result without limit 1. 

I printed the queries with and without limit 1 for getRawOne method and observed that

With limit 1 I can see 

![image](https://github.com/bcgov/SIMS/assets/54600590/2a8cceaa-c581-4416-81b2-1c2b9a0e5e24)


And without limit one 


![image](https://github.com/bcgov/SIMS/assets/54600590/6b84a107-494b-48df-bd39-a6191f21287a)


Hence without limit1 I am not sure what is the result returned and felt safe to use limit 1.

Reply:
Thanks for checking that.